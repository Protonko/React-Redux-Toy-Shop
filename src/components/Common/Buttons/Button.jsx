import React from 'react';
import PropTypes from 'prop-types';

export const Button = ({title, onClick, classNames = 'button'}) => {
  return (
    <button className={classNames} onClick={onClick}>
      {title}
    </button>
  );
}

Button.propTypes = {
  title: PropTypes.string,
  onClick: PropTypes.func,
};
