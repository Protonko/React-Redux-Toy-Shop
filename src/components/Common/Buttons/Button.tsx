// types
import {TClassName} from 'models/types';

import React, {FC} from 'react';
import classNames from 'classnames';

interface IProps {
  title: string,
  onClick: () => void,
  isRemoved?: boolean,
  disabled?: boolean,
  classNamesAdditional: TClassName,
}

export const Button: FC<IProps> = ({
  title,
  onClick,
  isRemoved,
  disabled,
  classNamesAdditional = []
}) => {
  const classNamesButton: string = classNames(
    'button',
    {'button--disabled': disabled},
    {'button--red': isRemoved},
    ...classNamesAdditional,
  );

  return (
    <button className={classNamesButton} onClick={onClick}>
      {title}
    </button>
  );
}
