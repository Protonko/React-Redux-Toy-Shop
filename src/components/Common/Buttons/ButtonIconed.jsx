import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

export const ButtonIconed = ({
  children,
  onClick,
  disabled,
  classNamesAdditional = [],
  ...attrs
}) => {
  const classNamesButton = classNames(
    'button',
    'button--iconed',
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button className={classNamesButton} onClick={onClick} {...attrs}>
      {children}
    </button>
  );
}

ButtonIconed.propTypes = {
  onClick: PropTypes.func,
  disabled: PropTypes.bool,
  classNamesAdditional: PropTypes.array,
};
