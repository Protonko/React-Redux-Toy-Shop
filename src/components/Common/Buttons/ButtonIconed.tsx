// types
import {TClassName} from 'models/types';

import React, {FC, ReactNode} from 'react';
import classNames from 'classnames';

export interface IProps {
  children: ReactNode,
  onClick: () => void,
  disabled?: boolean,
  classNamesAdditional?: TClassName,
  attrs?: Array<string | number>,
}

export const ButtonIconed: FC<IProps> = ({
  children,
  onClick,
  disabled,
  classNamesAdditional = [],
  ...attrs
}) => {
  const classNamesButton = classNames(
    'button',
    'button--iconed',
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button className={classNamesButton} onClick={onClick} {...attrs}>
      {children}
    </button>
  );
}
