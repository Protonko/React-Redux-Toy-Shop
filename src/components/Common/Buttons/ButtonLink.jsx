import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const ButtonLink = ({
  text,
  disabled,
  isRemoved,
  onClick,
  classNamesAdditional = []
}) => {
  const classNamesButton = classNames(
    'button',
    'button--linked',
    {'button--colored-red': isRemoved},
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button className={classNamesButton} onClick={onClick}>
      {text}
    </button>
  )
}

ButtonLink.propTypes = {
  text: PropTypes.string,
  disabled: PropTypes.bool,
  isRemoved: PropTypes.bool,
  onClick: PropTypes.func,
  classNamesAdditional: PropTypes.array,
};
