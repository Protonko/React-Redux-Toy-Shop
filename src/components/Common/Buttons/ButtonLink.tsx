import React, {FC} from 'react';
import classNames from 'classnames';
import {TClassName} from 'models/types';

interface IButtonLinkProps {
  text: string,
  disabled?: boolean,
  isRemoved?: boolean,
  onClick: () => any, // TODO - убрать
  classNamesAdditional?: TClassName,
}

export const ButtonLink: FC<IButtonLinkProps> = ({
  text,
  disabled,
  isRemoved,
  onClick,
  classNamesAdditional = [],
}) => {
  const classNamesButton = classNames(
    'button',
    'button--linked',
    {'button--colored-red': isRemoved},
    {'button--disabled': disabled},
    ...classNamesAdditional,
  );

  return (
    <button className={classNamesButton} onClick={onClick} type="button">
      {text}
    </button>
  )
}
