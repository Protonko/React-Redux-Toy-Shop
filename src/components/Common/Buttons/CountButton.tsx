import React, {FC} from 'react';
import classNames from 'classnames';

export interface IProps {
  isMinus?: boolean,
  onClick: () => void,
}

export const CountButton: FC<IProps> = ({
  isMinus,
  onClick,
}) => {
  const classNamesButton: string = classNames(
      'button',
      'button--count',
      {'button--count-minus': isMinus}
  );

  return (
      <button
        className={classNamesButton}
        onClick={onClick}
        data-symbol={isMinus ? '-' : '+'}
      />
  );
};
