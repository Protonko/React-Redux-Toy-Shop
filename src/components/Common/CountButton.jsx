import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

export const CountButton = ({isMinus, onClick}) => {
  const classNamesButton = classNames(
      'button',
      'button--count',
      {'button--count-minus': isMinus}
  );

  return (
      <button className={classNamesButton} onClick={onClick} data-symbol={isMinus ? '-' : '+'} />
  );
};

CountButton.propTypes = {
  isMinus: PropTypes.bool,
  onClick: PropTypes.func,
};
