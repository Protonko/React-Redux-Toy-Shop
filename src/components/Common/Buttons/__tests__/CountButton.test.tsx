// types
import {IProps} from 'components/Common/Buttons/CountButton';

import React from 'react';
import {shallow} from 'enzyme';
import {ELEMENT_SELECTORS_BUTTON} from 'static/__test__/button';
import {CountButton} from 'components/Common/Buttons/CountButton';

const shallowCardComponent = (props: IProps) => (
  shallow(<CountButton {...props} />)
)

describe('CountButton', () => {
  let props: IProps;
  let fn: jest.Mock<void, []>;

  beforeEach(() => {
    fn = jest.fn(() => {});
    props = {
      onClick: fn,
      isMinus: true,
    };
  });

  it('Match snapshot', () => {
    const component = shallowCardComponent(props);

    expect(component).toMatchSnapshot();
  });

  it('Should add className "button--count-minus"', () => {
    const countButtonComponent = shallowCardComponent(props);
    const classNameButton = countButtonComponent
      .find(ELEMENT_SELECTORS_BUTTON.button)
      .hasClass(ELEMENT_SELECTORS_BUTTON.minusButton);

    expect(classNameButton).toEqual(true);
  });
});
