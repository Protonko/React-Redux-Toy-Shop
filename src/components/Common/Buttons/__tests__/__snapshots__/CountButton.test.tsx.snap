// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountButton Match snapshot 1`] = `
<button
  className="button button--count button--count-minus"
  data-symbol="-"
  onClick={[MockFunction]}
/>
`;
